homeassistant/components/light/mysensors.py

```diff
@@ -109,118 +100,258 @@ def device_state_attributes(self):
         return device_attr
 
     @property
+    def available(self):
+        """Return True if entity is available."""
+        return self.value_type in self._values
+
+    @property
     def is_on(self):
         """True if device is on."""
         return self._state
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
+        if self.gateway.optimistic:
+            # optimistically assume that light has changed state
+            self.update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        value_type = kwargs.get('value_type')
+        value = kwargs.get('value')
+        if value_type is not None and value is not None:
```

Comment:
I would update the code to be like this:
```python
if value_type is None or value is None:
    _LOGGER.warning(…)
    return
```
Will allow you to get rid of the `else`